sympy/core/basic.py

```diff
@@ -2041,6 +2052,12 @@ def could_extract_minus_sign(self):
         return False  # see Expr.could_extract_minus_sign
 
 
+# For all Basic subclasses _prepare_class_assumptions is called by
+# Basic.__init_subclass__ but that method is not called for Basic itself so we
+# call the function here instead.
+_prepare_class_assumptions(Basic)
```

Comment:
I'm not sure why this is better than `_BasicBase` though, we can simply inherit `Printable -> _BasicBase -> Basic` inheritance instead of using multiple inheritance.

Reply:
I guess it was addressed in the comments though.

Reply:
A redundant class in the MRO is a significant complexity that it is better to avoid. It has runtime implications as well as being something that many users and contributors will see and will need to understand what the redundant class is doing.